docs/content/preview/explore/ysql-language-features/data-manipulation.md

```diff
@@ -1,386 +1,371 @@
+When a NULL value for name is inserted, you will get an error as:
 
-```sql
-INSERT INTO employees (name, department) VALUES ('John Smith', 'Sales');
+```sql{.nocopy}
+ERROR:  23502: null value in column "name" violates not-null constraint
 ```
 
-Alternatively, you can provide the `DEFAULT` keyword as the column's value, as shown in the following example:
-
-```sql
-INSERT INTO employees (employee_no, name, department)
-VALUES (DEFAULT, 'John Smith', 'Sales');
-```
+### Foreign key constraint
 
-When you create your sequence via  `serial` , the sequence has all its parameters set to default values. For example, the sequence would not be optimized for access to its information because it does not have a cache (the default value of the a `SEQUENCE` 's  `CACHE` parameter is 1;  `CACHE`  defines how many sequence numbers should be pre-allocated and stored in memory). To be able to configure a sequence at the time of its creation, you need to construct it explicitly and then reference it when you create your table, as shown in the following examples:
+Foreign Key constraints are used to enforce referential integrity between two tables in a relational database. They create a link between data in two tables, ensuring that relationships between tables remain consistent. Foreign keys help prevent orphaned records by ensuring that references to related data remain valid. For example, consider the scenario where you have a `departments` table that has `dept_id` and other info and an employee table that links the employee with `dept_id`.
 
 ```sql
-CREATE SEQUENCE sec_employees_employee_no
-START 1
-CACHE 1000;
-```
+CREATE TABLE departments (
+    dept_id INT PRIMARY KEY,
+    dept_name VARCHAR(100)
+);
 
-```sql
 CREATE TABLE employees (
-    employee_no integer DEFAULT nextval('sec_employees_employee_no') NOT NULL,
-    name text,
-    department text
+    emp_id INT PRIMARY KEY,
+    emp_name VARCHAR(100),
+    dept_id INT,
+        CONSTRAINT fk_department FOREIGN KEY (dept_id)
+            REFERENCES departments(dept_id)
 );
 ```
 
-The new sequence value is generated by the `nextval()` function.
-
-## Update rows
-
-YSQL allows you to update a single row in table, all rows, or a set of rows. You can update each column separately.
-
-If you know (1) the name of the table and column that require updating, (2) the rows that need to be modified, and (3) the new value for the column, you can use the `UPDATE`  statement in conjunction with the `SET`  clause to modify data, as shown in the following example:
+As you have a foreign key constraint set up between these 2 tables, you will not be able to insert a row into the `employees` tables with an invalid `dept_id`. It will throw a `violates foreign key constraint` error like:
```

Comment:
```suggestion
As you have a foreign key constraint set up between these tables, you won't be able to insert a row into the `employees` tables with an invalid `dept_id`. It will throw a `violates foreign key constraint` error:
```